### ISSUE TITLE
Strange behavior using modelFor from a nested engine route

### ISSUE THREAD

**Issue by jackson-dean** [ember-engines/ember-engines#592]
I have an engine which has an engine mounted in its routes.js. This nested engine is able to directly reference the route model of the parent engine, as long as there is no route in the child engine having the same name. For example, if I have routes/application.js in the parent engine, but do not have an application.js in the child engine I can directly access the parents application route via `this.modelFor('application')` from a route in the nested engine. I don't think this is the intended behavior, but I could be mistaken. I am curious is this is just an unintended sideeffect of how engines are implemented or if this is actually the way it is intended to work. We are using engines to lazy load routes more so than for isolation, so sharing state is something we want to do, but we don't want to rely on an unintended quirk that could get removed out from under us.

**Comment by DerekJacobi** [comment]
@jackson-dean have you looked here http://ember-engines.com/guide/mounting-engines ?, specifically about resetting the namespace? `Router.map(function() {
  this.mount('super-blog', { resetNamespace: true, path: '/blog' });
});`

**Comment by jackson-dean** [comment]
yeah, have been mulling over the guide for some time. I don't think resetNamespace really addresses the issue. I tried setting the option to true but still has the same behavior. If I don't export a `routes/application.js` from the child engine, then modelFor('application') from the child engine directly gives me the parent engine's application route model. I'm trying to understand if this is just a quirk or if this is expected and explicitly supported. There are other ways to share state (via the store, or some other service), so if that is the recommended way to communicate between parent engine and child engine, then I can do that. But just wanted to understand the intention for how it should be expected to work.

**Comment by DerekJacobi** [comment]
oh ok I see i misread what the initial question was, it seems like that would be how the engine should work, I will try playing around with that on my end

**Comment by DerekJacobi** [comment]
Also is this an in repo engine? Trying this in a separate engine does not work for me

**Comment by jackson-dean** [comment]
it is an in-repo engine. As soon as I have time, I will try to create a simple reproduction repo and share it with you.
